File: src/main/java/uk/gov/ons/ssdc/caseprocessor/model/dto/NewCase.java
package uk.gov.ons.ssdc.caseprocessor.model.dto;

import java.util.Map;
import java.util.UUID;
import lombok.Data;

@Data
public class NewCase {
  private UUID caseId;

  private UUID collectionExerciseId;

  private Map<String, String> sample;

  private Map<String, String> sampleSensitive;
}

File: python_spike/caseprocessor/dto/newCase.py
from dataclasses import dataclass
from typing import Dict
import uuid
from dataclasses_json import LetterCase, dataclass_json


@dataclass_json(letter_case=LetterCase.CAMEL)
@dataclass
class NewCase:
    case_id: uuid
    collection_exercise_id: uuid
    sample: Dict[str, str]
    sample_sensitive: Dict[str, str]
